packages/core/src/Services/socket-general.js

```diff
@@ -106,7 +106,7 @@ const BinarySocketGeneral = (() => {
                         if (!mt5_list_response.error) {
                             client_store.responseMt5LoginList(mt5_list_response);
                             WS.balanceAll().then((balance_response) => {
-                                this.setBalance(balance_response.balance);
+                                if (!balance_response.error) client_store.setBalance(balance_response.balance);
```

Comment:
1. Does this render the `setBalance` function in `this` useless? If so, can we delete that?
2. Can we call `this.setBalance` instead under that condition? As it also calls `client_store.setBalance` under it.

Reply:
1. `setBalance` is used by `ResponseHandlers` so we shouldn't delete it.
2. We actually don't need to `yield` to `website_status` as we already have the `balanceAll` response in this call, so I directly passed it to`client_store.setBalance` instead.